pkg/controller/certificate-shim/helper.go

```diff
@@ -92,5 +96,14 @@ func translateAnnotations(crt *cmapi.Certificate, ingLikeAnnotations map[string]
 		}
 		crt.Spec.Usages = newUsages
 	}
+
+	if revisionHistoryLimit, found := ingLikeAnnotations[cmapi.RevisionHistoryLimitAnnotationKey]; found {
+		limit, err := strconv.ParseInt(revisionHistoryLimit, 10, 32)
+		if err != nil {
+			return fmt.Errorf("%w %q: %v", errInvalidIngressAnnotation, cmapi.RevisionHistoryLimitAnnotationKey, err)
+		}
+		crt.Spec.RevisionHistoryLimit = pointer.Int32(int32(limit))
+	}
```

Comment:
Here we should also validate that the value is not smaller than 1 and if not throw the same `errInvalidIngressAnnotation` error

Reply:
Yeah, thanks, done

Reply:
> if not throw the same errInvalidIngressAnnotation error

Apologies this was not precise- we should also tell the user that the value has to be 1 or more (as some folks may actually set it to 0). So maybe conctentate that error message with the value passed and a message that the value has to be > 1

Reply:
Added some details for this error case